File: spring-graalvm-native-samples/webmvc-tomcat/src/main/java/com/example/webmvc/WebmvcApplication.java
package com.example.webmvc;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication(proxyBeanMethods = false)
public class WebmvcApplication {

	public static void main(String[] args) {
		SpringApplication.run(WebmvcApplication.class, args);
	}

}

File: spring-graalvm-native-samples/webmvc-kotlin/src/main/kotlin/com/example/webmvc/WebmvcApplication.kt
package com.example.webmvc

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication(proxyBeanMethods = false)
class WebmvcApplication

fun main(args: Array<String>) {
	runApplication<WebmvcApplication>(*args)
}

data class Bar(val message: String)

@RestController
class WebmvcController {

	@GetMapping("/")
	fun greet(): Bar {
		return Bar("hi!")
	}
}
